garrysmod/gamemodes/sandbox/gamemode/commands.lua

```diff
@@ -774,11 +774,31 @@ function Spawn_SENT( ply, EntityName, tr )
 		if ( !EntTable ) then return end
 
 		PrintName = EntTable.PrintName
+		
+		-- If the player is aimed at an entity, do not place below the entity
+		if ( !tr.HitWorld ) then
+			tr.HitNormal.z = 0

```

Comment:
Modifying the trace data could screw up other things using the CanTool hook. Try this
```lua
local SpawnPos = tr.HitPos
if ( tr.HitWorld ) then
    SpawnPos  = SpawnPos + tr.HitNormal * 16
end
```

Reply:
Thank you!